src/api/general-services/pipeline-status.js

```diff
@@ -143,11 +144,13 @@ const getPipelineStatus = async (experimentId, processName) => {
         nextToken,
       }).promise();
 
+      if (execution.status === 'FAILED') {
+        error = history.events[history.events.length - 1].executionFailedEventDetails;
+      }
```

Comment:
This is probably not correct. You can't assume that the `Execution Failed` event will always be the last event in the list. In addition, in some cases it may be possible for multiple different activities to fail at the same time (e.g. if we use a Map). In any case, there is no contract that this is accurate. What I'd do is maybe put this on line L150, between `while` and `completedSteps`, but instead do a backwards iteration on the array until a proper event is found, something akin to:

```
let error = _.findLast(events, (elem) => elem.type === 'ExecutionFailed');

if(error) {
  error = error.executionFailedEventDetails;
}
```